assigner/backends/decorators.py

```diff
@@ -0,0 +1,15 @@
+from assigner.backends import GitlabBackend, MockBackend
+from assigner.config import config_context
+
+
+def require_backend(func):
+    """Provides a backend depending on configuration."""
+    @config_context
+    def wrapper(config, cmdargs, *args, **kwargs):
+        if config.backend == "gitlab":
+            return func(config, GitlabBackend, cmdargs, *args, **kwargs)
+        if config.backend == "mock":
+            return func(config, MockBackend, cmdargs, *args, **kwargs)
+
```

Comment:
Let's warn here so that users can fix their config.

Reply:
I will add a check for no backend option. However, Config enforces that backend must match the regex:

```
WARNING  Your configuration is not valid: 'mocks' does not match '^(mock|gitlab)?$'
```

Reply:
True. I will handle some of this in the config versioning PR which I'm going to do today.